language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -260,7 +260,7 @@ genModule pkgMap (Scope scope) curPkgId mod
                       (PackageRef, ModuleName) -> T.Text -> T.Text
     importDecl pkgMap modRef@(pkgRef, modName) rootPath =
       "import * as " <>  genModuleRef modRef <> " from '" <>
-      T.intercalate "/" ((rootPath : pkgRefStr pkgMap pkgRef : ["lib" | pkgRef /= PRSelf]) ++ unModuleName modName) <>
+      modPath ((rootPath : pkgRefStr pkgMap pkgRef : ["lib" | pkgRef /= PRSelf]) ++ unModuleName modName) <>
```

Comment:
This changes behavior: `joinPath` is platform dependent. On Windows it will produce backslashes, i.e., `joinPath ["a", "b"] == "a\\b"`. Does that still work?  If it does, did it work before, i.e., do both `/` and `\` work on Windows? If so, would it be better to use `/` consistently everywhere so the generated typescript code is platform independent?

Reply:
the approach we've taken in ghc-lib scripts is to use '/' even on windows.
- i'll change the implementation of `modPath` to use `intercalate` rather than `joinPath`
- `import System.FilePath.Posix((</>)) -- Make sure we generate / on all platforms.`
the change to `modPath`  has the additional benefit of cutting down on more `T.unpack`'s too.

Reply:
I know how it works for ghc-lib. What I don’t know if javascript import syntax imposes special restrictions. From looking around, I couldn’t find a clear answer to this.

Reply:
it would seem according to https://shapeshed.com/writing-cross-platform-node/ that forward slashes will work on windows.

Reply:
i expect the two suggested changes above is a good idea so as to negate the possibility of future maintainers wondering about it all over.

Reply:
That doesn’t talk about imports which is the thing I’m worried about. Anyway, given that basically all examples I can find use forward slashes that seems like a safe choice and it is platform independent which makes me happy since I like reproducible builds across platforms.

Reply:
@cocreature do you agree with this?

Reply:
> basically all examples I can find use forward slashes that seems like a safe choice 

agreed. have pushed a revision. good to go?